src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/WithdrawalTest.kt

```diff
@@ -1469,5 +1404,33 @@ class WithdrawalTest : IntegrationTestBase() {
     return booking
   }
 
-  private fun datePeriodForDuration(start: LocalDate, duration: Int) = DatePeriod(start, start.plusDays(duration.toLong()))
+  private fun nowPlusDays(days: Long) = LocalDate.now().plusDays(days)
+
+  private fun toDatePeriod(start: LocalDate, duration: Int) = DatePeriod(start, start.plusDays(duration.toLong()))
+
+  fun toWithdrawable(application: ApplicationEntity) = Withdrawable(
+    application.id,
+    WithdrawableType.application,
+    emptyList(),
+  )
+
+  fun toWithdrawable(placementRequest: PlacementRequestEntity) = Withdrawable(
+    placementRequest.id,
+    WithdrawableType.placementRequest,
+    listOf(toDatePeriod(placementRequest.expectedArrival, placementRequest.duration)),
+  )
+
+  fun toWithdrawable(placementApplication: PlacementApplicationEntity) =
+    Withdrawable(
+      placementApplication.id,
+      WithdrawableType.placementApplication,
+      placementApplication.placementDates.map { toDatePeriod(it.expectedArrival, it.duration) },
+    )
+
+  fun toWithdrawable(booking: BookingEntity) =
+    Withdrawable(
+      booking.id,
+      WithdrawableType.booking,
+      listOf(DatePeriod(booking.arrivalDate, booking.departureDate)),
+    )
```

Comment:
These functions are is really just for the tests only because i didn't want to depend on the WithdrawableTransformer, which is how we do this transform in the 'actual' code.

There is a related question as to whether we should introduce an interface to each Entities service for the 'getWithdrawableState' function, which they all implement. Whilst i think we could do that to make this common structure clear in code, because of the nature of how Cas1WithdrawableTreeBuilder uses these functions i don't think there's any syntactic advantage to doing this